File: clients/algoliasearch-client-php/lib/Model/Analytics/Direction.php
<?php

namespace Algolia\AlgoliaSearch\Model\Analytics;

/**
 * Direction Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class Direction
{
    /**
     * Possible values of this enum
     */
    const ASC = 'asc';

    const DESC = 'desc';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::ASC,
            self::DESC,
        ];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/analytics/Direction.java
package com.algolia.model.analytics;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Gets or Sets direction */
@JsonAdapter(Direction.Adapter.class)
public enum Direction {
  ASC("asc"),

  DESC("desc");

  private final String value;

  Direction(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static Direction fromValue(String value) {
    for (Direction b : Direction.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<Direction> {

    @Override
    public void write(final JsonWriter jsonWriter, final Direction enumeration)
      throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public Direction read(final JsonReader jsonReader) throws IOException {
      String value = jsonReader.nextString();
      return Direction.fromValue(value);
    }
  }
}
